test/unit/ngsiv2/expressions/expressionBasedTransformations-test.js

```diff
@@ -0,0 +1,763 @@
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/entities/ws1/attrs', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextRequests/updateContextExpressionPlugin4.json'))
+                .reply(204);
+        });
+
+        it('should calculate it and add it to the payload', function(done) {
+            iotAgentLib.update('ws1', 'WeatherStationMultiple', '', values, function(error) {
+                should.not.exist(error);
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When a measure arrives and there is not enough information to calculate an expression', function() {
+        // Case: Update for integer with expression that needs more attributes
+
+        var values = [
+            {
+                name: 'p',
+                type: 'Integer',
+                value: 52
+            }
+        ];
+
+        beforeEach(function() {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/entities/ws1/attrs', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextRequests/updateContextExpressionPlugin1.json'))
+                .reply(204);
+        });
+
+        it('should not calculate the expression', function(done) {
+            iotAgentLib.update('ws1', 'WeatherStation', '', values, function(error) {
+                should.not.exist(error);
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When an update comes for attributes without expressions and type integer', function() {
+        // Case: Update for an integer attribute without expression
+        var values = [
+            {
+                name: 'p',
+                type: 'Integer',
+                value: 52
+            }
+        ];
+
+        beforeEach(function() {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/entities/light1/attrs', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextRequests/updateContextExpressionPlugin11.json'))
+                .reply(204);
+        });
+
+        it('should apply the expression before sending the values', function(done) {
+            iotAgentLib.update('light1', 'Light', '', values, function(error) {
+                should.not.exist(error);
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When an update comes for attributes with numeric expressions and type integer', function() {
```

Comment:
Test case L364-L392 seems to be equal to test case L303-L332. Maybe I'm missing some difference?

Reply:
Fixed in 799b13427fa60951e8b67e6fb1d1df0e8402a89f

Reply:
I suppose that when we added more tests to cover all the possible cases, we duplicated this one.